### PR TITLE
Create /admin/stop-announcing resource to admin port

### PR DESCRIPTION
**Pull request by johngmyers**
Create /admin/stop-announcing resource to admin port

### CODE REVIEW

File: jmx-http/src/main/java/com/proofpoint/jmx/StopAnnouncingResource.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
```
Comment (gwittel) [inline jmx-http/src/main/java/com/proofpoint/jmx/StopAnnouncingResource.java]: 2015

File: jmx-http/src/main/java/com/proofpoint/jmx/StopAnnouncingResource.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        this.adminServerCredentialVerifier = requireNonNull(adminServerCredentialVerifier, "adminServerCredentialVerifier is null");
+    }
+
+    @PUT
+    public void stopAnnouncing(@HeaderParam("Authorization") String authHeader)
+    {
+        adminServerCredentialVerifier.authenticate(authHeader);
+
+        log.info("Received shutdown request. Stopping discovery announcer.");
+        announcer.destroy();
```
Comment (gwittel) [inline jmx-http/src/main/java/com/proofpoint/jmx/StopAnnouncingResource.java]: Do we care if multiple shutdown requests come in while one is in progress?  Seems like the Announcer won't care; (worst case, multiple de-announces)

Comment (johngmyers) [inline jmx-http/src/main/java/com/proofpoint/jmx/StopAnnouncingResource.java]: Don't think so. Multiple de-announces are fine, as that is an idempotent operation.

**Comment by gwittel** [comment]
LGTM

**Comment by johngmyers** [comment]
Rebased, squashed, and merged.

**Pull request closed without merging**